lib/types/local-time.js

```diff
@@ -27,257 +30,234 @@ const nanoSecInMillis = Long.fromNumber(1000000);
- * @returns {LocalTime}
- */
-LocalTime.now = function (nanoseconds) {
-    return LocalTime.fromDate(new Date(), nanoseconds);
-};
+        const regex = /^(?:[01]?\d|2[0-4]):[0-5]?\d:[0-5]?\d(?:\.\d{1,9})?$/;
+        if (!regex.test(value)) {
+            throw new TypeError(`Argument format is invalid`);
+        }
 
-/**
- * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
- * @param {Date} date Local date portion to extract the time passed since midnight.
- * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the nanosecond time portion.
- * @returns {LocalTime}
- */
-LocalTime.fromDate = function (date, nanoseconds) {
-    if (!util.isDate(date)) {
-        throw new Error("Not a valid date");
+        return new LocalTime(
+            bigintToLong(rust.LocalTimeWrapper.fromString(value)),
+        );
     }
-    //Use the local representation, only the milliseconds portion
-    const millis =
-        (date.getTime() + date.getTimezoneOffset() * -60000) % millisInDay;
-    return LocalTime.fromMilliseconds(millis, nanoseconds);
-};
 
-/**
- * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
- * @param {Number} milliseconds A Number from 0 to 86,399,999.
- * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
- * @returns {LocalTime}
- */
-LocalTime.fromMilliseconds = function (milliseconds, nanoseconds) {
-    if (typeof nanoseconds !== "number") {
-        nanoseconds = 0;
+    /**
+     * Uses the current local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Number} [nanoseconds] A Number from 0 to 999 999 999 representing the time nanosecond portion.
+     * @returns {LocalTime}
+     */
+    static now(nanoseconds) {
+        return LocalTime.fromDate(new Date(), nanoseconds);
     }
-    return new LocalTime(
-        Long.fromNumber(milliseconds)
-            .multiply(nanoSecInMillis)
-            .add(Long.fromNumber(nanoseconds)),
-    );
-};
 
-/**
- * Creates a new instance of LocalTime from the bytes representation.
- * @param {Buffer} value
- * @returns {LocalTime}
- */
-LocalTime.fromBuffer = function (value) {
-    if (!(value instanceof Buffer)) {
-        throw new TypeError("Expected Buffer, obtained " + util.inspect(value));
+    /**
+     * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Date} date Local date portion to extract the time passed since midnight.
+     * @param {Number} [nanoseconds] A Number from 0 to 999 999 999 representing the nanosecond time portion.
+     * @returns {LocalTime}
+     */
+    static fromDate(date, nanoseconds) {
+        if (typeof date === "Date") {
+            throw new TypeError("Not a valid date");
+        }
+        // Use the local representation, only the milliseconds portion
```

Comment:
Can you explain more here or reword the comment? I do not quite get what is done here.

Reply:
Oh, that's a comment from DataStax ...